recipes/openssl/1.x.x/conanfile.py

```diff
@@ -119,8 +119,8 @@ def build_requirements(self):
             if not self.options.no_asm and not tools.which("nasm"):
                 self.build_requires("nasm/2.14")
         if self._win_bash:
-            if "CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() != 'msys2':
```

Comment:
Why do you need msys2 if your subsystem is msys2? You can add it to your profile instead, if you want to force from Conan package.

Reply:
Without this patch building packages in github actions is not possible. I guess conan uses "git bash" as "msys2" environment.

Reply:
So I suggest you adding msys2 as build requirement to your profile. Currently, CCI uses Jenkins only.

Reply:
I don't understand what is a principal difference between this package and e.g. libxml2 which always installs msys2 for case when comipiler is not "Visual Studio": https://github.com/conan-io/conan-center-index/blob/master/recipes/libxml2/all/conanfile.py#L70

cc @SSE4

Reply:
as I remember, OpenSSL doesn't need MSYS for Visual Studio builds, only perl, as it's not really auto-tools based

Reply:
Yes that's correct. I guess main issue here is that Conan recognizes tools coming with git as a full-featured msys2 environment, and it's not enough here because of shrinked perl

Reply:
yes, I remember issues like that, if wrong perl is picked up build may fail with cryptic error message. we had to ensure strawberry perl (build requirement) is always used for Visual Studio. that's why we set `PERL` environment variable to point to the right interpreter instance.

Reply:
It says it doesn't have Pod::Usage and doesn't allow to install it from CPAN either

Reply:
I didn't actually find out which perl executable gets used but now I'm pretty sure it's one coming with git, as well as bash

Reply:
What I understand of this issue: on Windows, if `msys2` is not used as build requirement, we have to require `strawberryperl`.

So why not this (without comments) ?

```python
    def build_requirements(self):
        if tools.os_info.is_windows:
            if self._win_bash and not tools.get_env("CONAN_BASH_PATH"): # please give me a shell (and perl as a side effect)
                self.build_requires("msys2/20200517")
            else: # I don't need a shell (either because I have already one or because I use nmake) , but I need perl
                self.build_requires("strawberryperl/5.30.0.1")
            if not self.options.no_asm and not tools.which("nasm"):
                self.build_requires("nasm/2.14")
```

Then update `self._perl` function accordingly.

Reply:
Strawberry Perl should be used if and only if build is performed in cmd.exe shell, because it uses Windows-style paths. In your code if `CONAN_BASH_PATH` is defined, Strawberry Perl will be used in POSIX-like shell which is going to fail.

```python
    def build_requirements(self):
        if tools.os_info.is_windows:
            if self._win_bash:
                if not tools.get_env("CONAN_BASH_PATH"):
                    self.build_requires("msys2/20200517") # I don't trust your perl, give me a real one for use in bash
                # else: you supply your bash so if you have broken perl it's your trouble too
            else: #  I don't trust your perl, give me a real one for use in cmd.exe
                self.build_requires("strawberryperl/5.30.0.1")
            if not self.options.no_asm and not tools.which("nasm"):
                self.build_requires("nasm/2.14")
```

Reply:
>Then update self._perl function accordingly.

What do you mean? Current `self._perl` is working fine.